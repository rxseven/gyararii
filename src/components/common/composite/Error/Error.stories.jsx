import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';

import Error from './index';

const title = 'Components/Common/Composite/Error';
const props = { error: ['Error 1', 'Error 2'], onDismiss: action('onDismiss') };
const component = <Error {...props} />;

storiesOf(title, module).add('default', () => component);
