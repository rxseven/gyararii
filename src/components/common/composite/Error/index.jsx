import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

import Alert from 'components/common/base/Alert';

const propTypes = exact({
  error: PropTypes.arrayOf(PropTypes.string).isRequired,
  onDismiss: PropTypes.func.isRequired
});

function Error({ error, onDismiss }) {
  return (
    <If condition={error.length > 0}>
      {error.map(message => (
        <Alert key={`key-${message}`} onDismiss={onDismiss}>
          {message}
        </Alert>
      ))}
    </If>
  );
}

Error.propTypes = propTypes;

export default Error;
