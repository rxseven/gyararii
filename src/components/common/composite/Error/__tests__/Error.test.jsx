import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Error from '../index';

describe('<Error />', () => {
  // Arrange
  const props = { error: ['Error 1', 'Error 2'], onDismiss: jest.fn() };
  const component = <Error {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
