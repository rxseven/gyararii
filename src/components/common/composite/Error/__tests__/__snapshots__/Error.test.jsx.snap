// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> Snapshot tests should render correctly 1`] = `
Array [
  <div
    className="alert alert-danger"
  >
    Error 1
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
  </div>,
  <div
    className="alert alert-danger"
  >
    Error 2
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
  </div>,
]
`;
